app/routes/users.js

```diff
@@ -4,8 +4,15 @@ import { inject as service } from '@ember/service';
 
 export default class UsersRoute extends Route {
   @service store;
+  @service dataLoader;
+
 
   async model() {
+    // Load the tasks and users concurrently
+    await Promise.all([
+      this.dataLoader.loadUsers(),
+      this.dataLoader.loadTasks(),
+    ]);
     // Retrieve all users from the store
     let allUsers = this.store.peekAll('user');
 
```

Comment:
Added this in as the users page was not loading directly, say if it threw an error or you went to an invalid URL and then back to the users page it was not showing the tasks correctly under each user. 
Im not sure if this is the correct solution, but I decided that it would make sense to trigger the dataLoader service from the users route in case the data was not available to that route directly and it seems to have fixed it